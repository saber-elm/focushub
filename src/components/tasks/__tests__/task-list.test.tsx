import { fireEvent, render, screen } from '@testing-library/react'
import { describe, expect, it, vi } from 'vitest'
import { axe, toHaveNoViolations } from 'jest-axe'
import TaskList from '../task-list'

const mockTasks = [
  {
    id: '1',
    title: 'Task One',
    completed: false,
  },
  {
    id: '2',
    title: 'Task Two',
    completed: true,
  },
]

describe('TaskList', () => {
  it('renders a list of TaskItem components', () => {
    render(
      <TaskList tasks={mockTasks} onToggle={() => {}} onDelete={() => {}} />
    )
    expect(screen.getByText('Task One')).toBeInTheDocument()
    expect(screen.getByText('Task Two')).toBeInTheDocument()
  })

  it('displays a message when there are no tasks', () => {
    render(<TaskList tasks={[]} onToggle={() => {}} onDelete={() => {}} />)
    expect(screen.getByText('No tasks yet.')).toBeInTheDocument()
  })
})

describe('TaskList snapshots', () => {
  it('matches snapshot with tasks', () => {
    const { asFragment } = render(
      <TaskList tasks={mockTasks} onToggle={() => {}} onDelete={() => {}} />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('matches snapshot with no tasks', () => {
    const { asFragment } = render(
      <TaskList tasks={[]} onToggle={() => {}} onDelete={() => {}} />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})

expect.extend(toHaveNoViolations)

const accessibleMockTasks = [
  { id: '1', title: 'Accessible task', completed: false },
]

describe('TaskList Accessibility ', () => {
  it('has no accessibility violation', async () => {
    const { container } = render(
      <TaskList
        tasks={accessibleMockTasks}
        onToggle={() => {}}
        onDelete={() => {}}
      />
    )
    const results = await axe(container)
    expect(results).toHaveNoViolations()
  })
})
