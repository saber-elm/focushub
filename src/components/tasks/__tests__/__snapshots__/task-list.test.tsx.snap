// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TaskList snapshots > matches snapshot with no tasks 1`] = `
<DocumentFragment>
  <p
    class="text-gray-500"
  >
    No tasks yet.
  </p>
</DocumentFragment>
`;

exports[`TaskList snapshots > matches snapshot with tasks 1`] = `
<DocumentFragment>
  <ul
    class="space-y-2 mt-4"
  >
    <li>
      <div
        class="flex justify-between border px-4 py-2 rounded"
      >
        <span
          class="max-w-xl wrap-break-word min-w-1"
        >
          Task One
        </span>
        <div
          class="flex items-center gap-4"
        >
          <div
            class="space-x-1"
          >
            <input
              class="text-blue-600 hover:underline"
              data-testid="task-toggle"
              id="task-1"
              name="status"
              type="checkbox"
            />
            <label
              class="hover:underline"
              for="task-1"
            >
              inComplete
            </label>
          </div>
          <button
            class="text-red-500 hover:underline"
          >
            ❌ Delete
          </button>
        </div>
      </div>
    </li>
    <li>
      <div
        class="flex justify-between border px-4 py-2 rounded"
      >
        <span
          class="line-through text-gray-400"
        >
          Task Two
        </span>
        <div
          class="flex items-center gap-4"
        >
          <div
            class="space-x-1"
          >
            <input
              checked=""
              class="text-blue-600 hover:underline"
              data-testid="task-toggle"
              id="task-2"
              name="status"
              type="checkbox"
            />
            <label
              class="hover:underline"
              for="task-2"
            >
              Completed
            </label>
          </div>
          <button
            class="text-red-500 hover:underline"
          >
            ❌ Delete
          </button>
        </div>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
